### ISSUE TITLE
verify_xcode will okay tampered Xcode when GateKeeper disabled

### ISSUE THREAD

**Issue by adurdin** [KrauseFx/fastlane#620]
Using `spctl` to verify Xcode will not verify the code signature when GateKeeper is set to “Allow apps downloaded from Anywhere” (i.e. disabled):

    $ spctl --assess --verbose ~/Applications/Xcode-tampered.app
    /Users/andy/Applications/Xcode-tampered.app: accepted
    override=security disabled

Using `codesign --verify` will identify invalid signatures regardless of GateKeeper settings:

    $ codesign --verify ~/Applications/Xcode-tampered.app/
    /Users/andy/Applications/Xcode-tampered.app/: a sealed resource is missing or invalid

The `verify_xcode` command uses `codesign` only to gather signature metadata, not to verify the signature, so it will report success on a tampered Xcode when GateKeeper is disabled.

**Comment by KrauseFx** [comment]
@adurdin Thanks for submitting this issue and letting me know. 

So should `fastlane` use both GateKeeper (recommended by Apple) and `codesign --verify`? If so, this step will take around 5 minutes.

Correct me if I'm wrong, but since the `spctl` command returns `override=security disabled` the `fastlane` action `verify_xcode` will fail. 

How did you modify your Xcode to let your verification fail?

**Comment by adurdin** [comment]
No, I’m wrong: I overlooked the final `output.include?...` tests in `verify_xcode`. I saw that you were looking for “accepted” which is present—and missed the `source=…` which is not present. You’re right that `verify_xcode` will flag Xcode as invalid in this case.

The modification I made to Xcode was to (temporarily) rename `/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang` and put another file in its place—enough to invalidate the signature, and easy to restore.

Either way, certainly no need to use both `spctl --assess` and `codesign --verify`.
